src/modules/formBuilder/components/itemEditor/FormItemEditor.tsx

```diff
@@ -39,6 +40,9 @@ const FormItemEditor: React.FC<FormItemEditorProps> = ({
     // errors: errorState?.errors,
   });
 
+  // If typename is not present, then this is a new item that hasn't been saved yet.
+  const isNewItem = useMemo(() => !item.__typename, [item]);
```

Comment:
We could also make this a piece of state in `FormBuilder` that's set when a form builder palette button is pressed... might be a bit neater, but I could go either way

Reply:
I could go either way too. I opted for managing fewer pieces of state for simplicity, even though it feels a bit un-neat to rely on the presence of the typename, it does signify that the record was resolved from the backend so it should be persisted. We don't resolve the record on validation.